_functions.sh

```diff
@@ -494,7 +494,7 @@ initialize_product_settings() {
           else
             env_var PLF_BRANCH "${PRODUCT_BRANCH}"
           fi
-          env_var EXO_PROFILES "all"
+          env_var EXO_PROFILES "${EXO_PROFILES}"
```

Comment:
I didn't updated them all since some case seems to be done on purpose, I nevertheless agree that if  specified,  EXO_PROFILES need to take the value passed.
So : 
- setting the default value to "all"
- updating all EXO_PROFILES forced to all to "${EXO_PROFILES}"

Reply:
IMO the ``env_var EXO_PROFILES "${EXO_PROFILES}"`` lines can be completely removed as they are just changing a variable to its own value

Reply:
Once all redefinition are removed the profiles is nomore applied : 
[DEBUG]  EXO_PROFILES=all,No-Ad
2018-11-21 09:39:05,698 | INFO  | Command line argument: -Dexo.profiles=all [o.a.c.startup.VersionLoggerListener<main>] 
09:40:24.926 [localhost-startStop-2] INFO  org.elasticsearch.node.Node - JVM arguments ...-Dexo.profiles=all
in the descriptor I have this value : 
EXO_PROFILES=""
Besides adding : 
[ -z $EXO_PROFILES ] && EXO_PROFILES="${EXO_PROFILES}" in setenv-local.sh will result to this line to appear 
 [ -z $EXO_PROFILES ] && EXO_PROFILES=""
it seems EXO_PROFILES is empty in this line : (https://github.com/exoplatform/adt/blob/37f39242c463aecf5a537dab180c5e98f0a8f0ff/_functions.sh#L1306)

Reply:
I have tested locally with the removal of all the ``env_var EXO_PROFILES "${EXO_PROFILES}"`` lines.
My instance is well deployed with the right profile
```
2018-11-21 12:21:48,100 | INFO  | Command line argument: -Dexo.profiles=all,test [o.a.c.startup.VersionLoggerListener<main>]
```
The value is also well defined on the instance configuration : 
```
$ grep -i profile plfent-5.1.0-RC01-postgres.acceptance.exoplatform.org
EXO_PROFILES="all,test"
```
To test I have used this : 
```
export EXO_PROFILES=all,test
./adt-dev.sh undeploy -n plfent -v 5.1.0-RC01 -i postgres -p 200 -d DOCKER_POSTGRES:10
./adt-dev.sh deploy -n plfent -v 5.1.0-RC01 -i postgres -p 200 -d DOCKER_POSTGRES:10
./adt-dev.sh start -n plfent -v 5.1.0-RC01 -i postgres -p 200 -d DOCKER_POSTGRES:10
```
Did you have correctly undeploy and redeployed your instance when you have tested ?

Reply:
The root cause between the different results we're getting is that : 
- I use sudo (maybe miss with my env var) (sudo is required as docker is invoked )
- I use /etc/default/adt to set my env var (EXO_PROFILES in this particular case) as my env var are not passing.
How I was able to get same result as you : 
- Switched to root and exported my env var

Finally : 
Preceding to all redefinition removal